Exercises/numbers/numbers.js

```diff
@@ -0,0 +1,53 @@
+//1. Assign by hand same value to 3 variables in binary, octal and hexadecimal format.
+//const decimal = new Number('99');
+
+const binary = new Number('1100011');
+const octal = new Number('143');
+const hexadecimal = new Number(63);
+
+console.log(binary,octal,hexadecimal);
+
+//2. Implement function `extractNumbers` which will leave only numbers in passed string (in the beginning of the string).
+
+function extractNumbers () {
+  price = '999usd';
+  number = parseInt(price,10); 
+  console.log(number);
+}
+extractNumbers();
+
+//3. Declare a variable with decimal value and convert to binary, octal and hexadecimal.
+
+const numberDecimal = new Number('9');
+
+console.log((numberDecimal).toString(2)); //1001
+console.log((numberDecimal).toString(8)); //011
+console.log((numberDecimal).toString(16)); //9
+
+//4. Use `Math.random` to generate random value from `0` to `100`.
+
+let randomInteger = Math.floor(Math.random() * 101);
+console.log(randomInteger);
+
+//5. Make function which will take a array with numbers as a parameter and calculate average number and round to hundredths.
+
+function arrayCalculation () {
+  values = [101,202,303,404,505];
+  sum = values.reduce((previous,current) => current += previous);
+  average = sum / values.length; 
+  total =(Math.round(100*average)/100);
+  console.log(total);
+}
+arrayCalculation();
+
+//6. Generate array to pass to function above with values, and length generated by Math.random.
+
+let arrayValues = new Array(Math.floor(Math.random() * 100));
```

Comment:
better to use Array,form here, will be much more elegance but needs additional knowledge

Reply:
Sorry, I did not understand this part "better to use Array,form here,"
Should be here the Array used or not?

Reply:
The way you define your array is incorrect. Moreover, if you run it it will crash your browser tab.

1. If you create new array using `let myArray = new Array(length)` syntax it will be created with already predefined empty elements and length which you provided. And when you then push element to the array it's length increases.

![image](https://user-images.githubusercontent.com/8523446/66716354-8f546300-edd5-11e9-95f4-f450576694b7.png)

For example:

```js
const myArray = new Array(4); // [ empty x 4 ];
console.log(myArray.length); // 4

myArray.push('new element');
console.log(myArray.length); // 5

myArray.push('newer element');
console.log(myArray.length); // 6

console.log(myArray); // [empty × 4, "new element", "newer element"]
```

2.  On each iteration in you `for` loop the length of `arrayValues` increases so the `i` variable will never be less then `arrayValues.length`. So the loop you created is infinite.

Have you tested the code you wrote?

Reply:
The code would work if you set each elements by index instead of pushing:
```js
const arrayValues = new Array(Math.floor(Math.random() * 100));
for (let i = 0; i < arrayValues.length; i++) {
  arrayValues[i] = Math.floor(Math.random() * 100);
}
```

This code will work but it is still not compact enough.

Alternative ways to define an array of given length filled with random values are:

```js
const myArray1 = Array.from(Array(3), Math.random);
const myArray2 = Array(4).fill().map(Math.random);
```